packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt

```diff
@@ -63,17 +63,34 @@ abstract class AbstractMongoDbInspectionBridge(
                     cachedValue?.let {
                         val fileInExpression = PsiTreeUtil.getParentOfType(expression, PsiFile::class.java)
                         if (fileInExpression == null || fileInExpression.virtualFile == null) {
-                            inspection.visitMongoDbQuery(null, holder, cachedValue!!.value)
+                            inspection.visitMongoDbQuery(null, holder, cachedValue!!.value, dialect.formatter)
                         } else {
                             val dataSource =
                                 MongoDbVirtualFileDataSourceProvider().getDataSource(
                                     expression.project,
                                     fileInExpression.virtualFile,
                                 )
-                            inspection.visitMongoDbQuery(dataSource?.localDataSource, holder, cachedValue!!.value)
+                            inspection.visitMongoDbQuery(
+                                dataSource?.localDataSource,
+                                holder,
+                                cachedValue!!.value,
+                                dialect.formatter,
+                            )
                         }
                     }
                 }
             }
         }
 }
+
+/**
+ * Checks whether a provided problem description has already been registered with the ProblemsHolder for a given
+ * PsiElement
+ *
+ * @param problem - Description of the problem
+ * @param source - PsiElement for which the problem is to be checked
+ * @return Boolean
+ */
+fun ProblemsHolder.isProblemAlreadyRegistered(problem: String, source: PsiElement): Boolean = this.results.any {
```

Comment:
I'm fine keeping it for now but we should have a ticket to avoid doing this. If the issue is on the bridge, that for some reason is called twice, we will need a ticket to investigate why IntelliJ is doing this.

Reply:
Yea definitely - I will make a ticket for this and add a todo to remove this if unnecessary - linking to that ticket

Reply:
Created the following ticket for investigation - https://jira.mongodb.org/browse/INTELLIJ-60 and updated the doc with the ticket as well.